### PR TITLE
Remove repair nanobot idle power consumption

### LINKED ISSUE #45938: Repair Nanobots should switch themselves off when they are done
Or just cease consuming bionic power at least. Otherwise it only adds cumbersome micromanagement to avoid wasting bionic power at full health.

### PR DESCRIPTION
**Pull request by Hirmuolio**
#### Summary

SUMMARY: Balance "Remove repair nanobot idle power consumption"

#### Purpose of change

Closes #45938

Repair nanobots consume power even when they do not do any healing. 
The choice wether to keep them running is a non-choice. Nobody would ever conciously choose to keep them running for reason other than that they don't want to micromanage the CBM.

This also makes the nanobots unfeasible to be used at times when player has no full control such as sleeping. The nanobots would keep draining power until player woke up even after healing was finished.

This leaves the activation cost a bit pointless. But 80 J activation cost was only 2 seconds worth of healing running cost so it isn't significant.

#### Describe the solution

Remove idle cost from repair nanobots

#### Describe alternatives you've considered

Shut down the CBM when player is at full health.

Rework repair nanobots to work in such way that consciously turning them on/off is a meaningful choice. (delayed start, ramping up effect, multiple modes, etc.)

#### Testing

repair nanobot shows correct 40 J/t power consumption.
Repair nanobots still heal and consume 40 J/t power while healing.
Repair nanobots don't consume power when player is at full health.

### CODE REVIEW

**Comment by Zireael07** [comment]
Why not do what suggested in https://github.com/CleverRaven/Cataclysm-DDA/issues/45938#issuecomment-742210362 ?

Making it give SELF_AWARE when on gives a reason for the power drain, and/or making them shut off at full health would also solve the issue of draining power needlessly.

**Comment by Hirmuolio** [comment]
I decided against shutting down because I think keeping it on would be more common use case for player. Toggle option for "keep always on" and "shut down after heal" would maybe be better.

I don't think adding self aware to the CBM would help. The problem isn't exactly that you don't get benefit from constantly running the CBM. The problem is that you need to micromanage the CBM. Turn it on when you are damaged and then turn it off as soon as you are healed. The self aware addition wouldn't affect that at all.

But adding self aware to the CBM would still be good. The effect of "1 hp/minute" is opaque to the player. The game won't even tell you how much health a human has. Do I have 10 HP? 50 HP? 100 HP? Self aware would fix that.

File: src/bionics.cpp [hunk 1]
```diff
@@ -1561,13 +1561,12 @@ float Character::get_effective_efficiency( int b, float fuel_efficiency )
...
  * @param bio the bionic that is meant to be recharged.
  * @param amount the amount of power that is to be spent recharging the bionic.
- * @param factor multiplies the power cost per turn.
  * @return indicates whether we successfully charged the bionic.
  */
-static bool attempt_recharge( Character &p, bionic &bio, units::energy &amount, int factor = 1 )
+static bool attempt_recharge( Character &p, bionic &bio, units::energy &amount )
 {
     const bionic_data &info = bio.info();
-    units::energy power_cost = info.power_over_time * factor;
```
Comment (Fris0uman) [inline src/bionics.cpp:1570]: What was this thing doing and why are you removing it? It might very well be a good thing but this seem unrelated to the other changes

**Comment by Fris0uman** [comment]
This seems like an ok solution but it would be nice if we were to work on getting a more generic framework for "auto management of CBMs" something that would also work for https://github.com/CleverRaven/Cataclysm-DDA/issues/45642 But I don't have clear suggestions on that yet.

Also (it's completly out of scope here) but maybe it would be cool if the nanotbots required you to get some silicium or other similar raw material to replenish themselves it would add a bit more depth to the thing.

File: src/bionics.cpp (hunk 1 above)
Comment (Hirmuolio) [reply src/bionics.cpp:1570]: There is only one place where that function is called and the variable is always set to 1 (that is also removed).

It is unrelated. I just noticed it while looking at how normal power cost works and decided to throw it away while working here.

Probably old code for power cost modifiers based on environmental conditions, status effects, faulty bionics, etc. (at least that is what the comment says. The comment should probably be removed too).
